src/reader/tf_reader.py

```diff
@@ -73,38 +66,44 @@ def _tf_parse_function(self, serialized):
+        # Get the image as raw bytes.
+        image_raw = parsed_example['image']
+        dimension = tf.cast(parsed_example['size'], tf.int32).numpy()
+        # Decode the raw bytes so it becomes a tensor with type.
+        image_tensor = tf.io.decode_raw(image_raw, tf.float64)
+        size = dimension * dimension
+        dlp.update(image_size=size)
+        # image_tensor = tf.io.decode_image(image_raw)
+        resized_image = tf.reshape(image_tensor, [dimension, dimension])
+        return tf.pad(resized_image,
+                      ((0, self._args.max_dimension - dimension), (0, self._args.max_dimension - dimension)))
 
-    @event_logging(module=MY_MODULE, arguments=profile_args)
+    @dlp.log
     def next(self):
         _file_list = self._args.file_list_train if self.dataset_type is DatasetType.TRAIN else self._args.file_list_eval
         batch_size = self._args.batch_size if self.dataset_type is DatasetType.TRAIN else self._args.batch_size_eval
-        logging.debug(f"{utcnow()} Reading {len(_file_list)} files thread {self.thread_index} rank {self._args.my_rank}")
+        logging.debug(
+            f"{utcnow()} Reading {len(_file_list)} files thread {self.thread_index} rank {self._args.my_rank}")
         self._dataset = tf.data.TFRecordDataset(filenames=_file_list, buffer_size=self._args.transfer_size)
         self._dataset = self._dataset.shard(num_shards=self._args.comm_size, index=self._args.my_rank)
-        self._dataset = self._dataset.map(self._tf_parse_function, num_parallel_calls=self._args.computation_threads)
-        self._dataset = self._dataset.map(self._decode_image, num_parallel_calls=self._args.computation_threads)
+        self._dataset = self._dataset.map(
+            lambda x: tf.py_function(func=self.parse_image, inp=[x], Tout=[tf.float64])
+            , num_parallel_calls=self._args.computation_threads)
         self._dataset = self._dataset.batch(batch_size, drop_remainder=True)
-        total = math.ceil(len(_file_list) / batch_size)
-        count = 1
-        t0 = time()
-        for batch in self._dataset:
-            t1 = time()
-            count += 1
-            self.profile_args["step"] = count
-            PerfTrace.get_instance().event_complete(f"{self.next.__qualname__}.next", MY_MODULE, t0, t1 - t0, arguments=self.profile_args)
-            is_last = 0 if count <= total else 1
-            yield is_last, batch
-            t0 = time()
-
-    def read_index(self, index):
-        t0 = time()
-        val = super().read_index(index)
-        t1 = time()
-        PerfTrace.get_instance().event_complete(
-            f"{self.read_index.__qualname__}", MY_MODULE, t0, t1 - t0, arguments=self.profile_args)
-        return val
-
-    @event_logging(module=MY_MODULE, arguments=profile_args)
+        total = math.ceil(len(_file_list)/self._args.comm_size / batch_size)
+        step = 1
+        for batch in dlp.iter(self._dataset):
```

Comment:
Do u want to keep the dlp.iter only here?

Reply:
I updated the change and removed dlp.iter from Line 96 also.